# Importing libraries 

In [29]:
#Importing libraries 
import pandas as pd 
import numpy as np
import os 

path = r'C:\Users\ANITA BOADU\Achievement 4'

#Import orders product prior as 
df_ords_prior=pd.read_csv(os.path.join(path,'Data','Prepared Data','orders_products_prior.csv'),index_col=False)

#Import orders checked as 
df_ords=pd.read_csv(os.path.join(path,'Data','Prepared Data','orders_checked.csv'),index_col=False)

# Export data to pkl
df_merged_large.to_pickle(os.path.join(path, 'Data','Prepared Data', 'orders_products_combined.pkl'))

In [16]:
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

####### df_ords_prior.head()` shows the first five rows of the DataFrame so you can quickly preview what the data looks like. It's a simple way to check that the dataset loaded correctly.

In [18]:
df_ords.head()

Unnamed: 0  order_id  user_id eval_set  order_number  order_dow  \
0           1   2398795        1    prior             2          3   
1           2    473747        1    prior             3          3   
2           3   2254736        1    prior             4          4   
3           4    431534        1    prior             5          4   
4           5   3367565        1    prior             6          2   

   order_hour_of_day  days_since_prior_order  
0                  7                    15.0  
1                 12                    21.0  
2                  7                    29.0  
3                 15                    28.0  
4                  7                    19.0

####### df_ords.head()` displays the first five rows of the `df_ords` DataFrame, giving you a quick preview of its contents and structure.

In [19]:
df_ords_prior.shape

(32434489, 4)

#######`df_ords_prior.shape` returns the dimensions of the DataFrame.  
Here, `(32434489, 4)` means it has **32,434,489 rows** and **4 columns**

In [20]:
df_ords.shape

(3214874, 8)

#########'df_ords`returns the dimensions of the DataFrame.
Here, '(3214874, 8)' means it has **32,148,874 rows** and **8 columns**   

In [21]:
df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

######## This code merges `df_ords` and `df_ords_prior` using the column `order_id`, creating a new DataFrame called `df_merged_large`.  
The `indicator=True` option adds a column showing where each row came from (left, right, or both tables).

In [25]:
df_merged_large.head()

Unnamed: 0  order_id  user_id eval_set  order_number  order_dow  \
0           1   2398795        1    prior             2          3   
1           1   2398795        1    prior             2          3   
2           1   2398795        1    prior             2          3   
3           1   2398795        1    prior             2          3   
4           1   2398795        1    prior             2          3   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  7                    15.0         196                  1   
1                  7                    15.0       10258                  2   
2                  7                    15.0       12427                  3   
3                  7                    15.0       13176                  4   
4                  7                    15.0       26088                  5   

   reordered _merge  
0          1   both  
1          0   both  
2          1   both  
3          0   both  
4          1   both

####### `df_merged_large.head()` shows the first five rows of the merged DataFrame so you can quickly preview the result of the merge.

In [26]:
df_merged_large['_merge'].value_counts()

_merge
both          30356421
left_only            0
right_only           0
Name: count, dtype: int64

####### `df_merged_large['_merge'].value_counts()` counts how many rows came from the left DataFrame, the right DataFrame, or from both after the merge.

This output shows that all merged rows (`30,356,421`) appear in **both** DataFrames, and there are **no rows** found only in the left or right DataFrame.  
This means the merge was a perfect match on `order_id`.

####### Export data to pkl
df_merged_large.to_pickle(os.path.join(path, 'Data','Prepared Data', 'orders_products_combined.pkl'))

This line saves the merged DataFrame (`df_merged_large`) as a pickle file called **orders_products_combined.pkl** in the specified folder.  
Pickle format keeps the data exactly as it is so you can load it quickly later.